client/state/current-user/actions.js

```diff
@@ -23,3 +35,39 @@ export function setCurrentUserFlags( flags ) {
 		flags
 	};
 }
+
+export function uploadGravatar( file, bearerToken, email ) {
+	return dispatch => {
+		dispatch( { type: GRAVATAR_UPLOAD_START } );
+		const data = new FormData();
+		data.append( 'filedata', file );
+		data.append( 'account', email );
+		return request
+			.post( 'https://api.gravatar.com/v1/upload-image' )
+			.send( data )
+			.set( 'Authorization', 'Bearer ' + bearerToken )
+			.set( 'Accept-Language', '*' )
+			.then( () => {
+				dispatch( successNotice(
```

Comment:
Thanks for the link @aduth! I forgot we had that middleware. Let's go with this approach @v18 vs adding in extra state to keep track of errors/success.